system-addon/content-src/components/TopSites/TopSite.jsx

```diff
@@ -386,10 +402,11 @@ export class _TopSiteList extends React.PureComponent {
       dispatch: props.dispatch,
       intl: props.intl
     };
+    let phIndex = 0;
     for (let i = 0, l = props.TopSitesCount; i < l; i++) {
       const link = topSites[i];
       const slotProps = {
-        key: i,
+        key: link ? link.guid || link.url : `ph${phIndex++}`,
```

Comment:
We don't need a separate variable to track the hole index as if we have a link, it'll have a key, and if we don't have a link, it can just be `i`, yeah?

Reply:
I had it as `i` initially for placeholders and previews would get stuck once you hovered over a placeholder. But it's possible this isn't needed anymore with the other changes made. I'll verify.

Reply:
OK. yeah, so if I switch this to be `key: link ? link.guid || link.url : i,` the previews get stuck after hovering over one or two placeholders... only on linux though! wheeee

Reply:
We don't need a special `"ph0"` string, so just plain `phIndex++` is enough. Also, let's name it `holeIndex` and add a comment to where it's declared as to why this is necessary.

Separately, links don't have `guid`s I believe? Were you trying to differentiate between pinned links of the same url?